src/commands/service/create.ts

```diff
@@ -10,328 +10,144 @@ import {
   addNewService as addNewServiceToBedrockFile,
   YAML_NAME as BedrockFileName
 } from "../../lib/bedrockYaml";
+import { build as buildCmd, exit as exitCmd } from "../../lib/commandBuilder";
 import {
   addNewServiceToMaintainersFile,
   generateDockerfile,
   generateGitIgnoreFile,
   generateStarterAzurePipelinesYaml
 } from "../../lib/fileutils";
 import { checkoutCommitPushCreatePRLink } from "../../lib/gitutils";
+import { isPortNumber } from "../../lib/validator";
 import { logger } from "../../logger";
 import { IBedrockFileInfo, IHelmConfig, IUser } from "../../types";
+import decorator from "./create.decorator.json";
+
+export interface ICommandOptions {
+  displayName: string;
+  gitPush: boolean;
+  helmChartChart: string;
+  helmChartRepository: string;
+  helmConfigBranch: string;
+  helmConfigGit: string;
+  helmConfigPath: string;
+  k8sBackend: string;
+  maintainerEmail: string;
+  maintainerName: string;
+  middlewares: string;
+  packagesDir: string;
+  pathPrefix: string;
+  pathPrefixMajorVersion: string;
+  k8sBackendPort: string;
+}
+
+export interface ICommandValues extends ICommandOptions {
+  k8sPort: number;
+  middlewaresArray: string[];
+  variableGroups: string[];
+}
+
+export const fetchValues = (opts: ICommandOptions) => {
+  if (!isPortNumber(opts.k8sBackendPort)) {
+    throw new Error("value for --k8s-service-port is not a value port number");
```

Comment:
typo - maybe you meant `valid port number`?

Reply:
accept